Wait for response keyword fails when it get a list

**Describe the bug**
After I updated to version 19.0.0 the `Wait for response` keyword fails when it get a list back in the response. This did not happen in version 18.9.1

**To Reproduce**
Link to slack conversation https://robotframework.slack.com/archives/C015KB1QSDN/p1732620419276269

Stacktrace:
```
Traceback (most recent call last):
  File "C:\Python 3.9.10\lib\site-packages\Browser\playwright.py", line 170, in grpc_channel
    yield playwright_pb2_grpc.PlaywrightStub(self._channel)
  File "C:\Python 3.9.10\lib\site-packages\Browser\keywords\network.py", line 140, in _wait_for_http_response
    return _format_response(response_json)
  File "C:\Python 3.9.10\lib\site-packages\Browser\keywords\network.py", line 35, in _format_response
    _jsonize_content(response, "body")
  File "C:\Python 3.9.10\lib\site-packages\Browser\keywords\network.py", line 54, in _jsonize_content
    data[bodykey] = json.loads(data[bodykey])
  File "C:\Python 3.9.10\lib\json\__init__.py", line 339, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python 3.9.10\lib\site-packages\Browser\browser.py", line 1241, in run_keyword
    raise e
  File "C:\Python 3.9.10\lib\site-packages\Browser\browser.py", line 1230, in run_keyword
    return DynamicCore.run_keyword(self, name, args, kwargs)
  File "C:\Python 3.9.10\lib\site-packages\robotlibcore.py", line 173, in run_keyword
    return self.keywords[name](*args, **(kwargs or {}))
  File "C:\Python 3.9.10\lib\site-packages\Browser\keywords\promises.py", line 265, in wait_for_all_promises
    self.wait_for(*self.unresolved_promises)
  File "C:\Python 3.9.10\lib\site-packages\Browser\keywords\promises.py", line 250, in wait_for
    return [promise.result() for promise in promises]
  File "C:\Python 3.9.10\lib\site-packages\Browser\keywords\promises.py", line 250, in <listcomp>
    return [promise.result() for promise in promises]
  File "C:\Python 3.9.10\lib\concurrent\futures\_base.py", line 439, in result
    return self.__get_result()
  File "C:\Python 3.9.10\lib\concurrent\futures\_base.py", line 391, in __get_result
    raise self._exception
  File "C:\Python 3.9.10\lib\concurrent\futures\thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Python 3.9.10\lib\site-packages\Browser\keywords\network.py", line 247, in wait_for_response
    response = self._wait_for_http_response(matcher, timeout)
  File "C:\Python 3.9.10\lib\site-packages\Browser\keywords\network.py", line 140, in _wait_for_http_response
    return _format_response(response_json)
  File "C:\Python 3.9.10\lib\contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "C:\Python 3.9.10\lib\site-packages\Browser\playwright.py", line 177, in grpc_channel
    raise AssertionError(str(error))
AssertionError: the JSON object must be str, bytes or bytearray, not list
```

**Expected behavior**
`Wait for response` keyword can handle lists.

--- a/Browser/keywords/network.py
+++ b/Browser/keywords/network.py
@@ -41,16 +41,17 @@ def _format_response(response: dict) -> dict:
 
 
 def _jsonize_content(data, bodykey):
-    headers = json.loads(data["headers"])
+    headers = json.loads(data.get("headers", "{}"))
     data["headers"] = headers
     lower_headers = {k.lower(): v for k, v in headers.items()}
     if (
         "content-type" in lower_headers
         and "application/json" in lower_headers["content-type"]
+        and bodykey in data
         and data[bodykey]
     ):
         with contextlib.suppress(json.decoder.JSONDecodeError):
-            if not isinstance(data[bodykey], dict):
+            if not isinstance(data[bodykey], (dict, list)):
                 data[bodykey] = json.loads(data[bodykey])
 
 

--- a/atest/library/certificate.py
+++ b/atest/library/certificate.py
@@ -5,6 +5,7 @@ from cryptography.hazmat.primitives.serialization import load_pem_private_key
 from cryptography import x509
 from cryptography.x509.oid import NameOID
 
+
 def generate_ca_certificate(cert_path: str, key_path: str):
 
     # generate CA key
@@ -12,57 +13,64 @@ def generate_ca_certificate(cert_path: str, key_path: str):
 
     # write CA key to the file
     with open(key_path, "wb") as f:
-        f.write(ca_key.private_bytes(
-            encoding=serialization.Encoding.PEM,
-            format=serialization.PrivateFormat.PKCS8,
-            encryption_algorithm=serialization.NoEncryption(),
-        ))
+        f.write(
+            ca_key.private_bytes(
+                encoding=serialization.Encoding.PEM,
+                format=serialization.PrivateFormat.PKCS8,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+        )
 
-    subject = issuer = x509.Name([
-        x509.NameAttribute(NameOID.COMMON_NAME, "CA"),
-    ])
+    subject = issuer = x509.Name(
+        [
+            x509.NameAttribute(NameOID.COMMON_NAME, "CA"),
+        ]
+    )
 
     # generate CA cert
-    ca_cert = x509.CertificateBuilder().subject_name(
-        subject
-    ).issuer_name(
-        issuer
-    ).public_key(
-        ca_key.public_key()
-    ).serial_number(
-        x509.random_serial_number()
-    ).not_valid_before(
-        datetime.datetime.now(datetime.timezone.utc)
-    ).not_valid_after(
-        datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=2)
-    ).add_extension(
-        x509.BasicConstraints(ca=True, path_length=None),
-        critical=True,
-    ).add_extension(
-        x509.KeyUsage(
-            digital_signature=True,
-            content_commitment=False,
-            key_encipherment=False,
-            data_encipherment=False,
-            key_agreement=False,
-            key_cert_sign=True,
-            crl_sign=True,
-            encipher_only=False,
-            decipher_only=False,
-        ),
-        critical=True,
-    ).add_extension(
-        x509.SubjectKeyIdentifier.from_public_key(ca_key.public_key()),
-        critical=False,
-    ).sign(ca_key, hashes.SHA256())
+    ca_cert = (
+        x509.CertificateBuilder()
+        .subject_name(subject)
+        .issuer_name(issuer)
+        .public_key(ca_key.public_key())
+        .serial_number(x509.random_serial_number())
+        .not_valid_before(datetime.datetime.now(datetime.timezone.utc))
+        .not_valid_after(
+            datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=2)
+        )
+        .add_extension(
+            x509.BasicConstraints(ca=True, path_length=None),
+            critical=True,
+        )
+        .add_extension(
+            x509.KeyUsage(
+                digital_signature=True,
+                content_commitment=False,
+                key_encipherment=False,
+                data_encipherment=False,
+                key_agreement=False,
+                key_cert_sign=True,
+                crl_sign=True,
+                encipher_only=False,
+                decipher_only=False,
+            ),
+            critical=True,
+        )
+        .add_extension(
+            x509.SubjectKeyIdentifier.from_public_key(ca_key.public_key()),
+            critical=False,
+        )
+        .sign(ca_key, hashes.SHA256())
+    )
 
     # write CA cert to the file
     with open(cert_path, "wb") as f:
         f.write(ca_cert.public_bytes(serialization.Encoding.PEM))
 
 
-
-def generate_server_certificate(server_san: str, cert_path: str, key_path: str, ca_cert_path: str, ca_key_path: str):
+def generate_server_certificate(
+    server_san: str, cert_path: str, key_path: str, ca_cert_path: str, ca_key_path: str
+):
 
     # load CA key
     with open(ca_key_path, "rb") as f:
@@ -79,72 +87,88 @@ def generate_server_certificate(server_san: str, cert_path: str, key_path: str, 
 
     # write server key to the file
     with open(key_path, "wb") as f:
-        f.write(server_key.private_bytes(
-            encoding=serialization.Encoding.PEM,
-            format=serialization.PrivateFormat.PKCS8,
-            encryption_algorithm=serialization.NoEncryption(),
-        ))
+        f.write(
+            server_key.private_bytes(
+                encoding=serialization.Encoding.PEM,
+                format=serialization.PrivateFormat.PKCS8,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+        )
 
-    subject = x509.Name([
-        x509.NameAttribute(NameOID.COMMON_NAME, server_san),
-    ])
+    subject = x509.Name(
+        [
+            x509.NameAttribute(NameOID.COMMON_NAME, server_san),
+        ]
+    )
 
     # generate server cert
-    server_cert = x509.CertificateBuilder().subject_name(
-        subject
-    ).issuer_name(
-        ca_cert.subject
-    ).public_key(
-        server_key.public_key()
-    ).serial_number(
-        x509.random_serial_number()
-    ).not_valid_before(
-        datetime.datetime.now(datetime.timezone.utc)
-    ).not_valid_after(
-        datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=1)
-    ).add_extension(
-        x509.SubjectAlternativeName([
-            x509.DNSName(server_san),
-        ]),
-        critical=False,
-    ).add_extension(
-        x509.BasicConstraints(ca=False, path_length=None),
-        critical=True,
-    ).add_extension(
-        x509.KeyUsage(
-            digital_signature=True,
-            content_commitment=False,
-            key_encipherment=True,
-            data_encipherment=False,
-            key_agreement=False,
-            key_cert_sign=False,
-            crl_sign=True,
-            encipher_only=False,
-            decipher_only=False,
-        ),
-        critical=True,
-    ).add_extension(
-        x509.ExtendedKeyUsage([
-            x509.ExtendedKeyUsageOID.SERVER_AUTH,
-        ]),
-        critical=False,
-    ).add_extension(
-        x509.SubjectKeyIdentifier.from_public_key(server_key.public_key()),
-        critical=False,
-    ).add_extension(
-        x509.AuthorityKeyIdentifier.from_issuer_subject_key_identifier(
-            ca_cert.extensions.get_extension_for_class(x509.SubjectKeyIdentifier).value
-        ),
-        critical=False,
-    ).sign(ca_key, hashes.SHA256())
+    server_cert = (
+        x509.CertificateBuilder()
+        .subject_name(subject)
+        .issuer_name(ca_cert.subject)
+        .public_key(server_key.public_key())
+        .serial_number(x509.random_serial_number())
+        .not_valid_before(datetime.datetime.now(datetime.timezone.utc))
+        .not_valid_after(
+            datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=1)
+        )
+        .add_extension(
+            x509.SubjectAlternativeName(
+                [
+                    x509.DNSName(server_san),
+                ]
+            ),
+            critical=False,
+        )
+        .add_extension(
+            x509.BasicConstraints(ca=False, path_length=None),
+            critical=True,
+        )
+        .add_extension(
+            x509.KeyUsage(
+                digital_signature=True,
+                content_commitment=False,
+                key_encipherment=True,
+                data_encipherment=False,
+                key_agreement=False,
+                key_cert_sign=False,
+                crl_sign=True,
+                encipher_only=False,
+                decipher_only=False,
+            ),
+            critical=True,
+        )
+        .add_extension(
+            x509.ExtendedKeyUsage(
+                [
+                    x509.ExtendedKeyUsageOID.SERVER_AUTH,
+                ]
+            ),
+            critical=False,
+        )
+        .add_extension(
+            x509.SubjectKeyIdentifier.from_public_key(server_key.public_key()),
+            critical=False,
+        )
+        .add_extension(
+            x509.AuthorityKeyIdentifier.from_issuer_subject_key_identifier(
+                ca_cert.extensions.get_extension_for_class(
+                    x509.SubjectKeyIdentifier
+                ).value
+            ),
+            critical=False,
+        )
+        .sign(ca_key, hashes.SHA256())
+    )
 
     # write server cert to the file
     with open(cert_path, "wb") as f:
         f.write(server_cert.public_bytes(serialization.Encoding.PEM))
 
 
-
-def generate_client_certificate(client_cn: str, cert_path: str, key_path: str, ca_cert_path: str, ca_key_path: str):
+def generate_client_certificate(
+    client_cn: str, cert_path: str, key_path: str, ca_cert_path: str, ca_key_path: str
+):
 
     # load CA key
     with open(ca_key_path, "rb") as f:
@@ -161,59 +185,71 @@ def generate_client_certificate(client_cn: str, cert_path: str, key_path: str, c
 
     # write client key to the file
     with open(key_path, "wb") as f:
-        f.write(client_key.private_bytes(
-            encoding=serialization.Encoding.PEM,
-            format=serialization.PrivateFormat.PKCS8,
-            encryption_algorithm=serialization.NoEncryption(),
-        ))
+        f.write(
+            client_key.private_bytes(
+                encoding=serialization.Encoding.PEM,
+                format=serialization.PrivateFormat.PKCS8,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+        )
 
-    subject = x509.Name([
-        x509.NameAttribute(NameOID.COMMON_NAME, client_cn),
-    ])
+    subject = x509.Name(
+        [
+            x509.NameAttribute(NameOID.COMMON_NAME, client_cn),
+        ]
+    )
 
     # generate client cert
-    client_cert = x509.CertificateBuilder().subject_name(
-        subject
-    ).issuer_name(
-        ca_cert.subject
-    ).public_key(
-        client_key.public_key()
-    ).serial_number(
-        x509.random_serial_number()
-    ).not_valid_before(
-        datetime.datetime.now(datetime.timezone.utc)
-    ).not_valid_after(
-        datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=1)
-    ).add_extension(
-        x509.BasicConstraints(ca=False, path_length=None),
-        critical=True,
-    ).add_extension(
-        x509.KeyUsage(
-            digital_signature=True,
-            content_commitment=False,
-            key_encipherment=True,
-            data_encipherment=False,
-            key_agreement=False,
-            key_cert_sign=False,
-            crl_sign=True,
-            encipher_only=False,
-            decipher_only=False,
-        ),
-        critical=True,
-    ).add_extension(
-        x509.ExtendedKeyUsage([
-            x509.ExtendedKeyUsageOID.CLIENT_AUTH,
-        ]),
-        critical=False,
-    ).add_extension(
-        x509.SubjectKeyIdentifier.from_public_key(client_key.public_key()),
-        critical=False,
-    ).add_extension(
-        x509.AuthorityKeyIdentifier.from_issuer_subject_key_identifier(
-            ca_cert.extensions.get_extension_for_class(x509.SubjectKeyIdentifier).value
-        ),
-        critical=False,
-    ).sign(ca_key, hashes.SHA256())
+    client_cert = (
+        x509.CertificateBuilder()
+        .subject_name(subject)
+        .issuer_name(ca_cert.subject)
+        .public_key(client_key.public_key())
+        .serial_number(x509.random_serial_number())
+        .not_valid_before(datetime.datetime.now(datetime.timezone.utc))
+        .not_valid_after(
+            datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=1)
+        )
+        .add_extension(
+            x509.BasicConstraints(ca=False, path_length=None),
+            critical=True,
+        )
+        .add_extension(
+            x509.KeyUsage(
+                digital_signature=True,
+                content_commitment=False,
+                key_encipherment=True,
+                data_encipherment=False,
+                key_agreement=False,
+                key_cert_sign=False,
+                crl_sign=True,
+                encipher_only=False,
+                decipher_only=False,
+            ),
+            critical=True,
+        )
+        .add_extension(
+            x509.ExtendedKeyUsage(
+                [
+                    x509.ExtendedKeyUsageOID.CLIENT_AUTH,
+                ]
+            ),
+            critical=False,
+        )
+        .add_extension(
+            x509.SubjectKeyIdentifier.from_public_key(client_key.public_key()),
+            critical=False,
+        )
+        .add_extension(
+            x509.AuthorityKeyIdentifier.from_issuer_subject_key_identifier(
+                ca_cert.extensions.get_extension_for_class(
+                    x509.SubjectKeyIdentifier
+                ).value
+            ),
+            critical=False,
+        )
+        .sign(ca_key, hashes.SHA256())
+    )
 
     # write client cert to the file
     with open(cert_path, "wb") as f:

--- a/atest/library/common.py
+++ b/atest/library/common.py
@@ -34,7 +34,12 @@ def start_test_server():
     return port
 
 
-def start_test_https_server(server_cert_path: str, server_key_path: str, ca_cert_path: str, mutual_tls: bool = False):
+def start_test_https_server(
+    server_cert_path: str,
+    server_key_path: str,
+    ca_cert_path: str,
+    mutual_tls: bool = False,
+):
     global SERVERS
     port = str(find_free_port())
 
@@ -45,18 +50,29 @@ def start_test_https_server(server_cert_path: str, server_key_path: str, ca_cert
 
     # This seems to be a very strange behaviour: if we start the server with absolute paths, it prepends
     # them with its own path and is unable to find the file. Therefore we have to count the relative path from its directory.
-    server_cert_path = os.path.relpath(os.path.abspath(server_cert_path), start = test_app_dir)
-    server_key_path = os.path.relpath(os.path.abspath(server_key_path), start = test_app_dir)
-    ca_cert_path = os.path.relpath(os.path.abspath(ca_cert_path), start = test_app_dir)
+    server_cert_path = os.path.relpath(
+        os.path.abspath(server_cert_path), start=test_app_dir
+    )
+    server_key_path = os.path.relpath(
+        os.path.abspath(server_key_path), start=test_app_dir
+    )
+    ca_cert_path = os.path.relpath(os.path.abspath(ca_cert_path), start=test_app_dir)
 
     print(test_app_path)
     process = Popen(
-        ["node", test_app_path,
-          "-p", port,
-          "-c", server_cert_path,
-          "-k", server_key_path,
-          "-C", ca_cert_path,
-          "-M" if mutual_tls else "-T"],
+        [
+            "node",
+            test_app_path,
+            "-p",
+            port,
+            "-c",
+            server_cert_path,
+            "-k",
+            server_key_path,
+            "-C",
+            ca_cert_path,
+            "-M" if mutual_tls else "-T",
+        ],
         text=True,
         cwd=str(root_dir),
     )
@@ -86,7 +102,6 @@ def stop_test_server(port: str):
         logger.warn(f"Server with port {port} not found")
 
 
-
 def get_current_scope_from_lib(keyword: FormatterKeywords) -> list:
     browser = BuiltIn().get_library_instance("Browser")
     stack = browser.scope_stack["assertion_formatter"].get()
@@ -99,4 +114,6 @@ def numbers_are_close(number1: int, number2, difference: int) -> bool:
     logger.info(f"Numbers difference is {size_difference}")
     if size_difference < difference:
         return True
-    raise ValueError(f"Numbers differece is {size_difference}, but it should have been {difference}")
+    raise ValueError(
+        f"Numbers differece is {size_difference}, but it should have been {difference}"
+    )

--- a/atest/library/os_wrapper.py
+++ b/atest/library/os_wrapper.py
@@ -13,18 +13,28 @@ from robot.libraries.BuiltIn import BuiltIn  # type: ignore
 from robot.utils import timestr_to_secs  # type: ignore
 
 
-def wait_file_count_in_directory(path: str, count: int, pattern: Optional[str] = None, absolute: bool = False, timeout: timedelta = timedelta(seconds=3)):
+def wait_file_count_in_directory(
+    path: str,
+    count: int,
+    pattern: Optional[str] = None,
+    absolute: bool = False,
+    timeout: timedelta = timedelta(seconds=3),
+):
     wait_time = time.monotonic() + timestr_to_secs(timeout.total_seconds())
     while wait_time > time.monotonic():
         try:
-            file_count = len(OperatingSystem().list_files_in_directory(path, pattern, absolute))
+            file_count = len(
+                OperatingSystem().list_files_in_directory(path, pattern, absolute)
+            )
         except RuntimeError:
             file_count = None
         logger.info(f"File count in {path} is {file_count}")
         if file_count == count:
             return file_count
         time.sleep(0.42)
-    raise AssertionError(f"File count was {file_count}, but should have been {count} within {timeout} seconds")
+    raise AssertionError(
+        f"File count was {file_count}, but should have been {count} within {timeout} seconds"
+    )
 
 
 def wait_until_file_exists(path: Path, timeout: timedelta = timedelta(seconds=3)):
@@ -37,7 +47,7 @@ def wait_until_file_exists(path: Path, timeout: timedelta = timedelta(seconds=3)
 
 
 def glob_files(path: str) -> list:
-    """Returns files path.glob(**/*). """
+    """Returns files path.glob(**/*)."""
     files = Path(path).glob("**/*")
     find_files = [str(file.absolute()) for file in files if file.is_file()]
     logger.info(f"Files: \"{', '.join(find_files)}\"")
@@ -54,6 +64,7 @@ def get_enty_command() -> str:
     if bool(int(os.environ.get("SYS_VAR_CI_INSTALL_TEST", 0))):
         return "rfbrowser"
     return f"{sys.executable} -m Browser.entry"
+
 
 def verify_translation(filename: Path) -> dict:
     """Verifies translation file."""
@@ -77,6 +88,7 @@ def modify_sha256(filename: Path, *kw_names):
             data[kw]["sha256"] = str(random.randint(1, 1000))
     with filename.open("w") as file:
         json.dump(data, file, indent=4)
+
 
 def remove_sha256(filename: Path, *kw_names):
     with filename.open("r") as file:
@@ -114,6 +126,7 @@ def add_kw(filename: Path, *kw_names):
     with filename.open("w") as file:
         json.dump(data, file, indent=4)
 
+
 def is_macos() -> bool:
     return sys.platform == "darwin"
 
@@ -132,10 +145,16 @@ def _parse_fi_date(date: str) -> datetime:
     except ValueError:
         return datetime.strptime(date, "%Y-%m-%d %H:%M:%S")
 
+
 ROBOT_LIBRARY_CONVERTERS = {datetime: _parse_fi_date}
 
 
-def delta_time_is_less_than(expected_time: datetime, locator: str, max_difference: timedelta = timedelta(seconds=30), timeout: timedelta = timedelta(seconds=30)):
+def delta_time_is_less_than(
+    expected_time: datetime,
+    locator: str,
+    max_difference: timedelta = timedelta(seconds=30),
+    timeout: timedelta = timedelta(seconds=30),
+):
     """Fail if the difference between time and time in selector is greater than difference within timeout."""
     browser = BuiltIn().get_library_instance("Browser")
     time_page = _parse_fi_date(browser.get_text(locator))
@@ -147,12 +166,19 @@ def delta_time_is_less_than(expected_time: datetime, locator: str, max_differenc
             raise AssertionError(f"Time difference is greater than {max_difference}")
         time.sleep(0.42)
 
-def _delta_time_is_less_than(time1: datetime, time2: datetime, max_difference: timedelta = timedelta(seconds=30)) -> bool:
+
+def _delta_time_is_less_than(
+    time1: datetime, time2: datetime, max_difference: timedelta = timedelta(seconds=30)
+) -> bool:
     """Fail if the difference between time1 and time2 is greater than difference."""
-    logger.info(f"time1: {time1}, time2: {time2}, max difference: {max_difference.seconds}")
+    logger.info(
+        f"time1: {time1}, time2: {time2}, max difference: {max_difference.seconds}"
+    )
     time_difference = abs(time1.timestamp() - time2.timestamp())
     if time_difference > max_difference.seconds:
-        logger.info(f"Time difference {time_difference} is greater than {max_difference.seconds}")
+        logger.info(
+            f"Time difference {time_difference} is greater than {max_difference.seconds}"
+        )
         return False
     return True
 

--- a/atest/library/screenshot.py
+++ b/atest/library/screenshot.py
@@ -20,23 +20,25 @@ def get_pixel_color(img_path: str, x: int, y: int) -> Tuple[int, int]:
     """Returns the color as (r,g,b,a) Tuple[int]"""
     im: Image.Image = Image.open(img_path)
     pix = im.load()
-    return pix[x,y]
+    return pix[x, y]
 
 
-def compare_images(img1_path: str, img2_bytes: bytes, expect_failure: ExpectFailure = ExpectFailure.no):
+def compare_images(
+    img1_path: str, img2_bytes: bytes, expect_failure: ExpectFailure = ExpectFailure.no
+):
     """Returns True if the images are the same, False otherwise"""
-    im1: Image.Image = Image.open(img1_path).convert('RGB')
-    im2: Image.Image = Image.open(BytesIO(img2_bytes)).convert('RGB')
+    im1: Image.Image = Image.open(img1_path).convert("RGB")
+    im2: Image.Image = Image.open(BytesIO(img2_bytes)).convert("RGB")
     diff: Image.Image = ImageChops.difference(im1, im2)
     buffered = BytesIO()
     diff.save(buffered, format="PNG")
     img_str = base64.b64encode(buffered.getvalue()).decode()
     logger.info(
-            '</td></tr><tr><td colspan="3">'
-            '<img alt="screenshot" class="robot-seleniumlibrary-screenshot" '
-            f'src="data:image/png;base64,{img_str}" width="900px">',
-            html=True,
-        )
+        '</td></tr><tr><td colspan="3">'
+        '<img alt="screenshot" class="robot-seleniumlibrary-screenshot" '
+        f'src="data:image/png;base64,{img_str}" width="900px">',
+        html=True,
+    )
     box = diff.getbbox()
     if box and expect_failure == ExpectFailure.no:
         logger.info(f"box: {box}, type({type(box)}), but no idea why this fails.")

--- a/atest/library/show_trace_tool.py
+++ b/atest/library/show_trace_tool.py
@@ -18,7 +18,7 @@ def run_rfbrowser_help():
             cwd=exec_dir,
             stderr=subprocess.STDOUT,
             stdout=file,
-            shell=True
+            shell=True,
         )
     time.sleep(2)
     help_text = out_file.read_text()
@@ -42,12 +42,14 @@ def start_show_trace(zip_file: str):
             cwd=exec_dir,
             stderr=subprocess.STDOUT,
             stdout=file,
-            shell=True
+            shell=True,
         )
     logger.info("Give process time to start")
     time.sleep(3)
     logger.info(f"Trace viewer output: {out_file.read_text()}")
-    assert process.returncode is None, "Process should be still running, but it was not."
+    assert (
+        process.returncode is None
+    ), "Process should be still running, but it was not."
     return process, out_file
 
 
@@ -57,7 +59,7 @@ def _check_trace_process(process: subprocess.Popen, out_file: Path):
         psutil.pid_exists(pid)
     except NoSuchProcess:
         logger.info(process.stdout)
-        raise 
+        raise
     proc = psutil.Process(pid)
     binary = False
     show_trace = False
@@ -84,7 +86,10 @@ def _check_trace_process(process: subprocess.Popen, out_file: Path):
             if "node" in child_proc.name().lower():
                 logger.info(child_proc.name())
                 node = True
-            if "chromium" in child_proc.name().lower() or "chrome" in child_proc.name().lower():
+            if (
+                "chromium" in child_proc.name().lower()
+                or "chrome" in child_proc.name().lower()
+            ):
                 logger.info(child_proc.name())
                 chromium = True
         if chromium and node:
@@ -107,4 +112,3 @@ def check_trace_process(prcess: subprocess.Popen, out_file: Path):
         logger.info(f"Trace file output: {out_file.read_text()}")
         time.sleep(1)
     raise ValueError("No valid trace process found")
-

--- a/atest/test/01_Browser_Management/keyword_banner.py
+++ b/atest/test/01_Browser_Management/keyword_banner.py
@@ -3,15 +3,28 @@ from Browser import Browser
 from assertionengine.assertion_engine import verify_assertion, AssertionOperator
 from typing import Optional
 
+
 def get_computed_banner_style():
     b: Browser = BuiltIn().get_library_instance("Browser")
-    return b.evaluate_javascript('!prefix body', "element => window.getComputedStyle(element,':before')")
+    return b.evaluate_javascript(
+        "!prefix body", "element => window.getComputedStyle(element,':before')"
+    )
 
-def get_real_page_source(operator: AssertionOperator = None, expected: Optional[str] = None):
+
+def get_real_page_source(
+    operator: AssertionOperator = None, expected: Optional[str] = None
+):
     b: Browser = BuiltIn().get_library_instance("Browser")
     return verify_assertion(b.get_page_source(), operator, expected)
 
-def get_banner_content(operator: AssertionOperator = None, expected: Optional[str] = None):
+
+def get_banner_content(
+    operator: AssertionOperator = None, expected: Optional[str] = None
+):
     style = get_computed_banner_style()
-    content = BuiltIn().evaluate(style['content']) if style['content'].startswith('"') else style['content']
+    content = (
+        BuiltIn().evaluate(style["content"])
+        if style["content"].startswith('"')
+        else style["content"]
+    )
     return verify_assertion(content, operator, expected)

--- a/atest/test/01_Browser_Management/tracing_groups.py
+++ b/atest/test/01_Browser_Management/tracing_groups.py
@@ -2,12 +2,16 @@ import json
 from pathlib import Path
 from typing import Union
 
+
 def get_trace_lines(path: Path) -> list[dict]:
-    with path.open(encoding='utf-8') as file:
-        return [ json.loads(line) for line in file.readlines() if '"type":"before"' in line ]
+    with path.open(encoding="utf-8") as file:
+        return [
+            json.loads(line) for line in file.readlines() if '"type":"before"' in line
+        ]
+
 
 def get_file_line(file: Union[Path, None], line: int, column: int) -> str:
     if file is None:
         return
-    with file.open(encoding='utf-8') as file:
-        return file.readlines()[line-1]
+    with file.open(encoding="utf-8") as file:
+        return file.readlines()[line - 1]

--- a/atest/test/08_Scope_Tests/scope_logger.py
+++ b/atest/test/08_Scope_Tests/scope_logger.py
@@ -4,17 +4,30 @@ from Browser import Browser
 from typing import Optional
 
 
-def log_all_scopes(exp_timeout: float, exp_retry_assertions_for: float, exp_strict_mode: bool, exp_selector_prefix: Optional[str] = None):
+def log_all_scopes(
+    exp_timeout: float,
+    exp_retry_assertions_for: float,
+    exp_strict_mode: bool,
+    exp_selector_prefix: Optional[str] = None,
+):
     b: Browser = BuiltIn().get_library_instance("Browser")
     timeout = b.scope_stack["timeout"].get()
     retry_assertions_for = b.scope_stack["retry_assertions_for"].get()
     strict_mode = b.scope_stack["strict_mode"].get()
     selector_prefix = b.scope_stack["selector_prefix"].get()
 
-    assert timeout == exp_timeout, f"timeout: {timeout} ({type(timeout)}) != {exp_timeout} ({type(exp_timeout)})"
-    assert retry_assertions_for == exp_retry_assertions_for, f"retry_assertions_for: {retry_assertions_for} ({type(retry_assertions_for)}) != {exp_retry_assertions_for} ({type(exp_retry_assertions_for)})"
-    assert strict_mode == exp_strict_mode, f"strict_mode: {strict_mode} ({type(strict_mode)}) != {exp_strict_mode} ({type(exp_strict_mode)})"
-    assert selector_prefix == exp_selector_prefix, f"selector_prefix: {selector_prefix} ({type(selector_prefix)}) != {exp_selector_prefix} ({type(exp_selector_prefix)})"
+    assert (
+        timeout == exp_timeout
+    ), f"timeout: {timeout} ({type(timeout)}) != {exp_timeout} ({type(exp_timeout)})"
+    assert (
+        retry_assertions_for == exp_retry_assertions_for
+    ), f"retry_assertions_for: {retry_assertions_for} ({type(retry_assertions_for)}) != {exp_retry_assertions_for} ({type(exp_retry_assertions_for)})"
+    assert (
+        strict_mode == exp_strict_mode
+    ), f"strict_mode: {strict_mode} ({type(strict_mode)}) != {exp_strict_mode} ({type(exp_strict_mode)})"
+    assert (
+        selector_prefix == exp_selector_prefix
+    ), f"selector_prefix: {selector_prefix} ({type(selector_prefix)}) != {exp_selector_prefix} ({type(exp_selector_prefix)})"
 
     logger.info(f"timeout: {timeout}")
     logger.info(f"retry_assertions_for: {retry_assertions_for}")

--- a/atest/test/09_Plugins/ExamplePlugin.py
+++ b/atest/test/09_Plugins/ExamplePlugin.py
@@ -41,7 +41,8 @@ class ExamplePlugin(LibraryComponent):
     def get_location_object(self) -> dict:
         """Returns the location object of the current page.
 
-        This keyword calles the python keyword `Evaluate Javascript` to get the location object."""
+        This keyword calles the python keyword `Evaluate Javascript` to get the location object.
+        """
         location_dict = self.library.evaluate_javascript(None, f"window.location")
         logger.info(f"Location object:\n {json.dumps(location_dict, indent=2)}")
         return DotDict(location_dict)

--- a/atest/test/10_retest/mylib.py
+++ b/atest/test/10_retest/mylib.py
@@ -2,32 +2,44 @@ from robot.libraries.BuiltIn import BuiltIn
 from pathlib import Path
 from Browser import Browser
 
-def create_context_with_string_type_recordvideodir_and_get_type_of_recordVideo_dir(video_path):
-    browser: Browser = BuiltIn().get_library_instance('Browser')
-    record_video={'dir': video_path}
+
+def create_context_with_string_type_recordvideodir_and_get_type_of_recordVideo_dir(
+    video_path,
+):
+    browser: Browser = BuiltIn().get_library_instance("Browser")
+    record_video = {"dir": video_path}
     browser.new_context(recordVideo=record_video)
-    url = BuiltIn().get_variable_value('${LOGIN_URL}')
+    url = BuiltIn().get_variable_value("${LOGIN_URL}")
     browser.new_page(url)
-    return str(type(record_video['dir']))
+    return str(type(record_video["dir"]))
 
-def create_context_with_path_type_recordvideodir_and_get_type_of_recordVideo_dir(video_path):
-    browser: Browser = BuiltIn().get_library_instance('Browser')
-    record_video={'dir': Path(video_path)}
+
+def create_context_with_path_type_recordvideodir_and_get_type_of_recordVideo_dir(
+    video_path,
+):
+    browser: Browser = BuiltIn().get_library_instance("Browser")
+    record_video = {"dir": Path(video_path)}
     browser.new_context(recordVideo=record_video)
-    url = BuiltIn().get_variable_value('${LOGIN_URL}')
+    url = BuiltIn().get_variable_value("${LOGIN_URL}")
     browser.new_page(url)
-    return str(type(record_video['dir']))
+    return str(type(record_video["dir"]))
 
-def create_persistent_context_with_string_type_recordvideodir_and_get_type_of_recordVideo_dir(video_path):
-    browser: Browser = BuiltIn().get_library_instance('Browser')
-    record_video={'dir': video_path}
-    url = BuiltIn().get_variable_value('${LOGIN_URL}')
-    browser.new_persistent_context(recordVideo=record_video, url=url)
-    return str(type(record_video['dir']))
 
-def create_persistent_context_with_path_type_recordvideodir_and_get_type_of_recordVideo_dir(video_path):
-    browser: Browser = BuiltIn().get_library_instance('Browser')
-    record_video={'dir': Path(video_path)}
-    url = BuiltIn().get_variable_value('${LOGIN_URL}')
+def create_persistent_context_with_string_type_recordvideodir_and_get_type_of_recordVideo_dir(
+    video_path,
+):
+    browser: Browser = BuiltIn().get_library_instance("Browser")
+    record_video = {"dir": video_path}
+    url = BuiltIn().get_variable_value("${LOGIN_URL}")
     browser.new_persistent_context(recordVideo=record_video, url=url)
-    return str(type(record_video['dir']))
+    return str(type(record_video["dir"]))
+
+
+def create_persistent_context_with_path_type_recordvideodir_and_get_type_of_recordVideo_dir(
+    video_path,
+):
+    browser: Browser = BuiltIn().get_library_instance("Browser")
+    record_video = {"dir": Path(video_path)}
+    url = BuiltIn().get_variable_value("${LOGIN_URL}")
+    browser.new_persistent_context(recordVideo=record_video, url=url)
+    return str(type(record_video["dir"]))

--- a/tasks.py
+++ b/tasks.py
@@ -605,7 +605,9 @@ def lint_python(c, fix=False):
     print("Run mypy:")
     c.run("mypy --exclude .venv --config-file Browser/mypy.ini Browser/ bootstrap.py")
     print("Run black:")
-    c.run("black --config Browser/pyproject.toml tasks.py Browser/ bootstrap.py")
+    c.run(
+        "black --config Browser/pyproject.toml tasks.py Browser/ bootstrap.py utest atest"
+    )
     print("Run ruff:")
     ruff_cmd = "ruff check --config Browser/pyproject.toml Browser/ bootstrap.py"
     if fix:

--- a/utest/robotframework_browser_translation_as_list/__init__.py
+++ b/utest/robotframework_browser_translation_as_list/__init__.py
@@ -4,12 +4,6 @@ from pathlib import Path
 def get_language() -> list:
     curr_dir = Path(__file__).parent.absolute()
     return [
-        {
-            "language": "eng",
-            "path": curr_dir / "translate_1.json"
-        },
-        {
-            "language": "swe",
-            "path": curr_dir / "translate_2.json"
-        }
+        {"language": "eng", "path": curr_dir / "translate_1.json"},
+        {"language": "swe", "path": curr_dir / "translate_2.json"},
     ]

--- a/utest/robotframework_browser_translation_fi/__init__.py
+++ b/utest/robotframework_browser_translation_fi/__init__.py
@@ -3,7 +3,4 @@ from pathlib import Path
 
 def get_language() -> dict:
     curr_dir = Path(__file__).parent.absolute()
-    return {
-        "language": "fi",
-        "path": curr_dir / "translate.json"
-    }
+    return {"language": "fi", "path": curr_dir / "translate.json"}

--- a/utest/robotframework_browser_translation_not_working/__init__.py
+++ b/utest/robotframework_browser_translation_not_working/__init__.py
@@ -3,7 +3,4 @@ from pathlib import Path
 
 def get_language_not_working() -> dict:
     curr_dir = Path(__file__).parent.absolute()
-    return {
-        "language": "fi",
-        "path": curr_dir / "translate.json"
-    }
+    return {"language": "fi", "path": curr_dir / "translate.json"}

--- a/utest/test_cookie.py
+++ b/utest/test_cookie.py
@@ -21,6 +21,7 @@ def test_cookie_as_dot_dict_negative_expiry(cookie: Cookie):
     data = cookie._cookie_as_dot_dict({"expires": epoch})
     assert data.expires == datetime.fromtimestamp(epoch, tz=timezone.utc)
 
+
 def test_expiry(cookie: Cookie):
     assert cookie._expiry("1") == 1
     assert cookie._expiry("1 000") == 1000

--- a/utest/test_network.py
+++ b/utest/test_network.py
@@ -45,17 +45,13 @@ def test_format_response_with_invalid_json_body():
 
 
 def test_format_response_with_no_body():
-    response = _format_response(
-        {"headers": '{"content-type": "application/json"}'}
-    )
+    response = _format_response({"headers": '{"content-type": "application/json"}'})
     assert "body" not in response
     assert response["headers"]["content-type"] == "application/json"
 
 
 def test_format_response_with_no_headers():
-    response = _format_response(
-        {"body": '{"key":"value1"}'}
-    )
+    response = _format_response({"body": '{"key":"value1"}'})
     assert response["body"] == '{"key":"value1"}'
     assert response["headers"] == {}
 
@@ -77,7 +73,10 @@ def test_format_response_with_json_list_body():
     assert response["headers"]["content-type"] == "application/json"
 
     response = _format_response(
-        {"headers": '{"content-type": "application/json"}', "body": json.loads(body_list)}
+        {
+            "headers": '{"content-type": "application/json"}',
+            "body": json.loads(body_list),
+        }
     )
     assert isinstance(response["body"], list)
     assert response["body"][0]["key"] == "value1"

--- a/utest/test_network.py
+++ b/utest/test_network.py
@@ -1,3 +1,4 @@
+import json
 from Browser.keywords.network import _format_response
 
 
@@ -13,6 +14,7 @@ def test_response_parsing_caps():
         {"headers": '{"Content-Type": "application/json"}', "body": '{"key":"value2"}'}
     )
     assert response["body"]["key"] == "value2"
+    assert response["headers"]["Content-Type"] == "application/json"
 
 
 def test_response_parsing_text():
@@ -23,6 +25,7 @@ def test_response_parsing_text():
         }
     )
     assert response["body"] == "{key:'value3'}"
+    assert response["headers"]["Content-Type"] == "text/html; charset=UTF-8"
 
 
 def test_empty_response():
@@ -30,3 +33,53 @@ def test_empty_response():
         {"headers": '{"Content-Type": "application/json"}', "body": None}
     )
     assert response["body"] == None
+    assert response["headers"]["Content-Type"] == "application/json"
+
+
+def test_format_response_with_invalid_json_body():
+    response = _format_response(
+        {"headers": '{"content-type": "application/json"}', "body": "{invalid: json}"}
+    )
+    assert response["body"] == "{invalid: json}"
+    assert response["headers"]["content-type"] == "application/json"
+
+
+def test_format_response_with_no_body():
+    response = _format_response(
+        {"headers": '{"content-type": "application/json"}'}
+    )
+    assert "body" not in response
+    assert response["headers"]["content-type"] == "application/json"
+
+
+def test_format_response_with_no_headers():
+    response = _format_response(
+        {"body": '{"key":"value1"}'}
+    )
+    assert response["body"] == '{"key":"value1"}'
+    assert response["headers"] == {}
+
+
+def test_format_response_with_empty_response():
+    response = _format_response({})
+    assert "body" not in response
+    assert response["headers"] == {}
+
+
+def test_format_response_with_json_list_body():
+    body_list = '[{"key":"value1"}, {"key":"value2"}]'
+    response = _format_response(
+        {"headers": '{"content-type": "application/json"}', "body": body_list}
+    )
+    assert isinstance(response["body"], list)
+    assert response["body"][0]["key"] == "value1"
+    assert response["body"][1]["key"] == "value2"
+    assert response["headers"]["content-type"] == "application/json"
+
+    response = _format_response(
+        {"headers": '{"content-type": "application/json"}', "body": json.loads(body_list)}
+    )
+    assert isinstance(response["body"], list)
+    assert response["body"][0]["key"] == "value1"
+    assert response["body"][1]["key"] == "value2"
+    assert response["headers"]["content-type"] == "application/json"

--- a/utest/test_python_usage.py
+++ b/utest/test_python_usage.py
@@ -21,14 +21,15 @@ def application_server():
     process = subprocess.Popen(
         ["node", "./node/dynamic-test-app/dist/server.js", "-p", "7272"]
     )
-    
+
     # Wait for server to bind to port
     import socket
+
     start_time = time.time()
     while time.time() - start_time < 10:
         sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         try:
-            result = sock.connect_ex(('localhost', 7272))
+            result = sock.connect_ex(("localhost", 7272))
             if result == 0:
                 sock.close()
                 break
@@ -40,10 +41,10 @@ def application_server():
     else:
         process.terminate()
         raise RuntimeError("Server failed to start within timeout")
-    
+
     # Give the server a moment to initialize after binding
     time.sleep(0.5)
-    
+
     yield
     process.terminate()
 
@@ -97,10 +98,12 @@ def atexit_register(monkeypatch):
     monkeypatch.setattr(atexit, "register", register)
     return register
 
+
 def test_playwright_lazy_initialization(browser):
     assert browser._playwright is None
     browser.get_browser_catalog()
     assert isinstance(browser.playwright, Browser.playwright.Playwright)
+
 
 def test_open_page_get_text(application_server, browser):
     browser.new_page("localhost:7272/dist/")
@@ -229,6 +232,7 @@ def test_promise_to_wait_for_elements_state_with_name_arguments(browser):
     browser.go_to(url="https://www.google.com")
     assert (promise.running() or promise.done()) is True
 
+
 @pytest.fixture()
 def browser_locator_handler(tmpdir):
     Browser.Browser._output_dir = tmpdir
@@ -237,6 +241,7 @@ def browser_locator_handler(tmpdir):
     browser = Browser.Browser(jsextension=str(extension))
     yield browser
     browser.close_browser("ALL")
+
 
 def test_custom_locator_handler(browser_locator_handler, application_server):
     browser_locator_handler.new_page("http://localhost:7272/overlay.html")

--- a/utest/test_rfbrowser_translate.py
+++ b/utest/test_rfbrowser_translate.py
@@ -1,6 +1,13 @@
 from approvaltests import verify_all
 
-from Browser.entry.translation import MISSING_CHECKSUM, _get_heading, _table_doc_updated, DOC_CHANGED, NO_LIB_KEYWORD, MISSING_TRANSLATION
+from Browser.entry.translation import (
+    MISSING_CHECKSUM,
+    _get_heading,
+    _table_doc_updated,
+    DOC_CHANGED,
+    NO_LIB_KEYWORD,
+    MISSING_TRANSLATION,
+)
 
 
 def test_heading():
@@ -8,19 +15,28 @@ def test_heading():
 
 
 def test_body_line():
-    verify_all("body", [
-        _table_doc_updated("new_page", 42, MISSING_TRANSLATION),
-        _table_doc_updated("this_is_long_keyword_which_is_42_chars_len", 42, MISSING_TRANSLATION),
-        _table_doc_updated("close", 42, DOC_CHANGED),
-        _table_doc_updated("new_page", 8, MISSING_TRANSLATION),
-        _table_doc_updated("click", 8, NO_LIB_KEYWORD),
-        _table_doc_updated("click", 8, DOC_CHANGED),
-        ]
+    verify_all(
+        "body",
+        [
+            _table_doc_updated("new_page", 42, MISSING_TRANSLATION),
+            _table_doc_updated(
+                "this_is_long_keyword_which_is_42_chars_len", 42, MISSING_TRANSLATION
+            ),
+            _table_doc_updated("close", 42, DOC_CHANGED),
+            _table_doc_updated("new_page", 8, MISSING_TRANSLATION),
+            _table_doc_updated("click", 8, NO_LIB_KEYWORD),
+            _table_doc_updated("click", 8, DOC_CHANGED),
+        ],
     )
+
 
 def test_full_long_kw_table():
     lines = _get_heading(42)
     lines.append(_table_doc_updated("new_page", 42, MISSING_CHECKSUM))
-    lines.append(_table_doc_updated("this_is_long_keyword_which_is_42_chars_len", 42, MISSING_TRANSLATION))
+    lines.append(
+        _table_doc_updated(
+            "this_is_long_keyword_which_is_42_chars_len", 42, MISSING_TRANSLATION
+        )
+    )
     lines.append(_table_doc_updated("close", 42, DOC_CHANGED))
     verify_all("all with long kw name", lines)

--- a/utest/test_translation.py
+++ b/utest/test_translation.py
@@ -17,12 +17,14 @@ def test_no_translation(browser: Browser):
     assert browser._get_translation(None) is None
     assert browser._get_translation(False) is None
 
+
 def test_provide_translation(browser: Browser):
     lang_plugin = "robotframework_browser_translation_fi"
     file_path = Path(__file__).parent / lang_plugin / "translate.json"
     received_path = browser._get_translation("fi")
     assert received_path == file_path, received_path
     assert browser._get_translation("wrong") is None
+
 
 def test_provide_translation_as_list(browser: Browser):
     lang_plugin = "robotframework_browser_translation_as_list"
@@ -46,12 +48,12 @@ def test_translated_kw_and_docs(browser: Browser):
 
     spec = browser.keywords_spec["1_session_storage_set_item"]
     assert spec.argument_specification
-    doc: str =  spec.documentation
+    doc: str = spec.documentation
     assert doc.startswith("1 Save data to session storage")
 
     spec = browser.keywords_spec["1_cancel_download"]
     assert spec.argument_specification
-    doc: str =  spec.documentation
+    doc: str = spec.documentation
     assert doc.startswith("1 Cancels an active download.")
 
 
@@ -59,5 +61,5 @@ def test_no_translation():
     browser = Browser(language=None)
     spec = browser.keywords_spec["cancel_download"]
     assert spec.argument_specification
-    doc: str =  spec.documentation
+    doc: str = spec.documentation
     assert doc.startswith("Cancels an active download.")

